sim/rogue/talents.go

```diff
@@ -605,27 +605,56 @@ func (rogue *Rogue) registerHonorAmongThieves() {
 		return
 	}
 
-	// FIXME: Implement a flow of combo points that simulates the party/raid crit procs
+	// In an ideal party, you'd probably get up to 6 ability crits/s (Rate = 600).
+	//  Survival Hunters, Enhancement Shamans, and Assassination Rogues are particularly good.
+	if rogue.Options.HonorOfThievesCritRate <= 0 {
+		rogue.Options.HonorOfThievesCritRate = 400
+	}
+
 	procChance := []float64{0, 0.33, 0.66, 1}[rogue.Talents.HonorAmongThieves]
 	comboMetrics := rogue.NewComboPointMetrics(core.ActionID{SpellID: 51701})
 	honorAmongThievesID := core.ActionID{SpellID: 51701}
 
+	icd := core.Cooldown{
+		Timer:    rogue.NewTimer(),
+		Duration: time.Second,
+	}
+
+	maybeProc := func(sim *core.Simulation) {
```

Comment:
The current model for the proc is that the ICD is triggered, even on a failed roll that gives no CP. It does not appear to do so in this snippet

Reply:
It's highly unlikely that ICD is invoked without an actual proc, since nothing else works this way. I'll test on PTR, but that's very tedious since CPs aren't logged.